### ISSUE TITLE
Support of HCL2 files

### ISSUE THREAD

**Issue by tbugfinder** [inspec/inspec#4733]
## Detailed Description

Testing of infrastructure depends on the given input. Hashicorp terraform 0.12 is a widely used tool to define and maintain infrastructure as code. Code is written in HCL2 syntax.
Inspec should be able to read such HCL2 file so that it can be used as an input for test-cases.

## Context

Use the defined HCL2 code as input for test cases.

**Comment by miah** [comment]
You might be interested in [InSpec Iggy](https://github.com/mattray/inspec-iggy)

That doesn't parse out HCL2, it parses tfstate.. but it might get you where you want.

**Comment by tbugfinder** [comment]
Well, truthful testing shouldn't use values created by the API with the API. In that sense Inspec Iggy could help as a temporary workaround however I'm still researching possibilities getting my HCL2 input as a reference.

**Comment by dwizzle204** [comment]
Yes I agree this could be helpful in creating tests against the state file itself. Typical use case is there aren’t azure inspec tests for every resource so in the interim I could test for values in the state file directly

**Comment by tbugfinder** [comment]
Well, if tests are built with the statefile, I'd lack comparing expected (hcl2 code) parameters with deployment. Therefore I think it's not what I'm looking for.
As an intermediate helper I'm using hcl2tojson python script however I'd appreciate native hcl2 support.

**Comment by dwizzle204** [comment]
Oh, yes then maybe we are thinking about different things. I was thinking more of almost a direct replacement for https://www.inspec.io/docs/reference/resources/json/ with hcl2

So your test would be similar to

```
describe hcl2({ command: 'terraform show'}) do
  its('resource_type') { should eq 'azurerm_virtual_machine' }
end
```

**Comment by tbugfinder** [comment]
keep
